R/tbl_cross.R

```diff
@@ -0,0 +1,163 @@
+#' Create a cross table of summary statistics
+#'
+#' Wrapper for [tbl_summary] to cross tabulate two variables
+#'
+#' @param data A data frame
+#' @param row A column name (quoted or unquoted) in data to be used for columns
+#' of cross table
+#' @param col A column name (quoted or unquoted) in data to be used for rows
+#' of cross table
+#' @param label List of formulas specifying row and column variables labels,
+#' e.g. `list(response ~ "Drug Response", stage ~ "Path T Stage")`. If a
+#' variable's label is not specified here, the label attribute
+#' (`attr(data$age, "label")`) is used.  If
+#' attribute label is `NULL`, the variable name will be used.
+#' @param statistic A statistic name in curly brackets to
+#' be replaced with the numeric statistic (see glue::glue).
+#' The default is `{n}`. If percent argument is `"column"`, `"row"`, or `"cell"`,
+#' default is `{n} ({p}%)`.
+#' @param percent Indicates the type of percentage to return if using the.
+#' `{p}` statistic. Must be one of
+#' `"column"`, `"row"`, or `"cell"`. Default is `"column"` when `{p}` is
+#' specified as `statistic`.
+#' @param missing Indicates whether to include counts of `NA` values in the table.
+#' Allowed values are `"no"` (never display NA values),
+#' `"ifany"` (only display if any NA values), and `"always"`
+#' (includes NA count row for all variables). Default is `"ifany"`.
+#' @family tbl_summary tools
+#' @author Karissa Whiting
+#' @export
+#' @return A `tbl_summary` object
+#' @examples
+#' tbl_cross_ex <-
+#'   trial[c("response", "trt")] %>%
+#'   tbl_cross(row = trt, col = response)
+#'
+#' @section Example Output:
+#' \if{html}{\figure{tbl_cross_ex.png}{options: width=50\%}}
+#'
+tbl_cross <- function(data,
+                      row = NULL,
+                      col = NULL,
+                      label = NULL,
+                      statistic = NULL,
+                      percent = c("none", "column", "row", "cell"),
+                      missing = c("ifany", "always", "no"),
+                      missing_text = "Unknown") {
+  row_str <- var_input_to_string(
+    data = data,
+    select_input = !!rlang::enquo(row),
+    arg_name = "row_str",
+    select_single = TRUE
+  )
+
+  col_str <- var_input_to_string(
+    data = data,
+    select_input = !!rlang::enquo(col),
+    arg_name = "col_str",
+    select_single = TRUE
+  )
+
+  # matching arguments ---------------------------------------------------------
+  missing <- match.arg(missing)
+  percent <- match.arg(percent)
+
+
+  # if no x and y provided, default to first two col_strumns of data -----------
+  if (is.null(row_str) & is.null(col_str)) {
```

Comment:
hmmmm, i am starting to think that perhaps we should just require the user to pass the variable names...how often do you really want the first two columns tabulated against one another?